clang/test/dpct/query_api_mapping/cuBLAS/blas.cu

```diff
@@ -0,0 +1,2045 @@
+// cublasDzasum-NEXT:   }
+// cublasDzasum-NEXT:   oneapi::mkl::blas::column_major::asum(*handle, n, (std::complex<double>*)x, incx, res_temp_ptr_ct{{[0-9]+}});
+// cublasDzasum-NEXT:   if(sycl::get_pointer_type(res, handle->get_context())!=sycl::usm::alloc::device && sycl::get_pointer_type(res, handle->get_context())!=sycl::usm::alloc::shared) {
+// cublasDzasum-NEXT:     handle->wait();
+// cublasDzasum-NEXT:     *res = *res_temp_ptr_ct{{[0-9]+}};
+// cublasDzasum-NEXT:     sycl::free(res_temp_ptr_ct{{[0-9]+}}, dpct::get_default_queue());
+// cublasDzasum-NEXT:   }
+
+// RUN: dpct --cuda-include-path="%cuda-path/include" --query-api-mapping=cublasZgemm3m | FileCheck %s -check-prefix=cublasZgemm3m
+// cublasZgemm3m: CUDA API:
+// cublasZgemm3m-NEXT:   cublasZgemm3m(
+// cublasZgemm3m-NEXT:       handle /*cublasHandle_t*/, transa /*cublasOperation_t*/,
+// cublasZgemm3m-NEXT:       transb /*cublasOperation_t*/, m /*int*/, n /*int*/, k /*int*/,
+// cublasZgemm3m-NEXT:       alpha /*const cuDoubleComplex **/, a /*const cuDoubleComplex **/,
+// cublasZgemm3m-NEXT:       lda /*int*/, b /*const cuDoubleComplex **/, ldb /*int*/,
+// cublasZgemm3m-NEXT:       beta /*const cuDoubleComplex **/, c /*cuDoubleComplex **/, ldc /*int*/);
+// cublasZgemm3m-NEXT: Is migrated to:
+// cublasZgemm3m-NEXT:   oneapi::mkl::blas::column_major::gemm(*handle, transa, transb, m, n, k, dpct::get_value(alpha, *handle), (std::complex<double>*)a, lda, (std::complex<double>*)b, ldb, dpct::get_value(beta, *handle), (std::complex<double>*)c, ldc);
+
+// RUN: dpct --cuda-include-path="%cuda-path/include" --query-api-mapping=cublasSspmv | FileCheck %s -check-prefix=cublasSspmv
+// cublasSspmv: CUDA API:
+// cublasSspmv-NEXT:   cublasSspmv(handle /*cublasHandle_t*/, upper_lower /*cublasFillMode_t*/,
+// cublasSspmv-NEXT:               n /*int*/, alpha /*const float **/, a /*const float **/,
+// cublasSspmv-NEXT:               x /*const float **/, incx /*int*/, beta /*const float **/,
+// cublasSspmv-NEXT:               y /*float **/, incy /*int*/);
+// cublasSspmv-NEXT: Is migrated to:
+// cublasSspmv-NEXT:   oneapi::mkl::blas::column_major::spmv(*handle, upper_lower, n, dpct::get_value(alpha, *handle), a, x, incx, dpct::get_value(beta, *handle), y, incy);
+
+// RUN: dpct --cuda-include-path="%cuda-path/include" --query-api-mapping=cublasCsrot | FileCheck %s -check-prefix=cublasCsrot
+// cublasCsrot: CUDA API:
+// cublasCsrot-NEXT:   cublasCsrot(handle /*cublasHandle_t*/, n /*int*/, x /*cuComplex **/,
+// cublasCsrot-NEXT:               incx /*int*/, y /*cuComplex **/, incy /*int*/,
+// cublasCsrot-NEXT:               c /*const float **/, s /*const float **/);
+// cublasCsrot-NEXT: Is migrated to:
+// cublasCsrot-NEXT:   oneapi::mkl::blas::column_major::rot(*handle, n, (std::complex<float>*)x, incx, (std::complex<float>*)y, incy, dpct::get_value(c, *handle), dpct::get_value(s, *handle));
```

Comment:
let's split the file to speed up the lit test.

Reply:
done